### PR TITLE
cloudinary

### PR DESCRIPTION
**Pull request by Nileshmechlin**
no public folder need

### CODE REVIEW

File: src/utils/cloudinary.ts [hunk 1]
```diff
@@ -27,9 +40,6 @@ export const deleteFromCloudinary = async (publicId: string) => {
     const result = await cloudinary.uploader.destroy(publicId);
     return result;
   } catch (err: any) {
-    if (err instanceof ApplicationError) {
-      throw new ApplicationError(err.statusCode, err.message);
-    }
     throw new ApplicationError(500, 'Internal server error while deleting from Cloudinary');
```
Comment (DigantMalaviya) [inline src/utils/cloudinary.ts]: use err value so get proper error msg

File: src/utils/cloudinary.ts [hunk 2]
```diff
@@ -8,17 +8,30 @@ cloudinary.config({
...
+      (error: UploadApiErrorResponse | undefined, result: UploadApiResponse | undefined) => {
+        if (error) {
+          console.error("Cloudinary upload error:", error); // Log the error
+          return reject(new ApplicationError(500, 'Error uploading to Cloudinary'));
+        }
+
+        // Check if result is defined
+        if (!result) {
+          console.error("No response from Cloudinary"); // Log if no result
+          return reject(new ApplicationError(500, 'No response from Cloudinary'));
```
Comment (DigantMalaviya) [inline src/utils/cloudinary.ts]: use err value so get proper error msg

File: src/utils/cloudinary.ts [hunk 3]
```diff
@@ -8,17 +8,30 @@ cloudinary.config({
...
-  }
+// Function to upload an image to Cloudinary directly from memory
+export const uploadToCloudinary = (file: Express.Multer.File, folder: string): Promise<UploadApiResponse> => {
+  return new Promise((resolve, reject) => {
+    const stream = cloudinary.uploader.upload_stream(
+      { folder: folder },
+      (error: UploadApiErrorResponse | undefined, result: UploadApiResponse | undefined) => {
+        if (error) {
+          console.error("Cloudinary upload error:", error); // Log the error
+          return reject(new ApplicationError(500, 'Error uploading to Cloudinary'));
```
Comment (DigantMalaviya) [inline src/utils/cloudinary.ts]: use err value so get proper error msg

**Comment by Nileshmechlin** [comment]
resolved the conversation

**Review by shivshankar12345 — APPROVED** [review]

**Pull request merged**